tsdb/_tools/dedupe.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+import json
+import sys
+
+def safeget(dct, *keys):
+    for key in keys:
+        try:
+            dct = dct[key]
+        except KeyError:
+            return None
+    return dct
+
+def generate_event_key(parsed_line):
+    return parsed_line['kubernetes']['event']['involved_object']['uid']
+
+def generate_state_container_key(parsed_line):
+    key = parsed_line['kubernetes']['container']['name']
+    key += parsed_line['kubernetes']['pod']['name']
+    key += parsed_line['kubernetes']['node']['name']
+    container_id = safeget(parsed_line, 'kubernetes', 'container', 'id')
```

Comment:
I wonder if `parsed_line.get('kubernetes', []).get('container', []).get('id')` is more normal here. Someone from the perf team will want to review all of this anyway and they should know better than me.

Reply:
this is what I was looking for

Reply:
pushed: c0e211ee305a15910b32aa2cae252f77dea9e900